Modules/socketmodule.c

```diff
@@ -3489,7 +3489,7 @@ sock_connect(PySocketSockObject *s, PyObject *addro)
     }
 
     res = internal_connect(s, SAS2SA(&addrbuf), addrlen, 1);
-    if (res < 0)
+    if (res == -1)
         return NULL;
```

Comment:
In this case, you could have an ERRNO set to -1, so you should instead check whether an exception was set or not. In addition, is it possilbe to test this behaviour (namely, use a custom errno set to -1 for instance)?

In addition, a note for `internal_connect` should say that `-1` does not necessarily mean an error and that the caller should check whether an exception is set or not afterwards.

In addition `socket.connect_ex` should be updated as well (and documented, both at the C level *and* the RST level) since it returns the error code.

Reply:
> that the caller should check whether an exception is set or not afterward

Specifically: it would be better to check `PyErr_Occurred()` rather than the ambigous `res`.

Reply:
Thanks for the review.
I tried my best to translate your comments in the patch, but Python internals are difficult for me.

The point about documenting socket.connect_ex behavior isn't clear to me. ATM it says:
```
The error indicator is ``0`` if the operation succeeded, otherwise the value of the :c:data:`errno` variable
```